File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right =nums.length-1;

        if(nums[left] <= nums[right]){
            return nums[left];
        }

        while(left<=right){
            int mid = (left+right)/2;
            if((mid+1<nums.length && nums[mid]>nums[mid+1])){
                return nums[mid+1];
            }
            else if((mid-1>=0 && nums[mid]<nums[mid-1])){
                return nums[mid];
            }
            else if(nums[mid]>=nums[left]){
                left=mid+1;
            }
            else {
                right=mid-1;
            }
        }

        return -1;
        
        
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        '''algorithm 
        i am planning to a bfs traversal where i travserese each level and 
        note the level of the tree until the queue becomes empty and process the nodes 
        '''
        #first check base case 
        if not root:
            return 0
        queue=[] #intializing queue 
        #push the root and level as one 
        queue.append((root ,1))
        maxlevel=1
        while queue:
            #pop the current element and level
            ele,level=queue.pop(0)
            maxlevel=max(maxlevel,level)
            if ele.left:
                queue.append((ele.left,level+1))
            if ele.right:
                queue.append((ele.right, level+1))

        return maxlevel
        
    
        